domain/consensus/processes/syncmanager/antipast.go

```diff
@@ -145,15 +145,15 @@ func (sm *syncManager) missingBlockBodyHashes(stagingArea *model.StagingArea, hi
 		lowHash = selectedChild
 	}
 	if !foundHeaderOnlyBlock {
-		if lowHash == highHash {
+		if lowHash.Equal(highHash) {
```

Comment:
I found a case where the error log line below was falsely triggered due to the wrong equality test